### DISCUSSION TITLE
What to do with @queues?

### DISCUSSION THREAD

**Discussion by filmaj** [architect/architect#1257] [General]
- existing implementation uses [short polling](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-short-and-long-polling.html), which blows through the AWS monthly free tier in ~10 days (due to the queue being hammered multiple times a second, checking if new messages have arrived)
- SNS has simpler semantics and now supports FIFO so can supplant `@queues` functionally

Suggestion: replace arc's `@queues` with FIFO-enabled SNS?

**Comment by trsrm** [comment]
At least, add a configuration option for that.

**Comment by ryanblock** [comment]
Switch over in Arc 10?

**Comment by dduran1967** [comment]
If I need SNS I can use @events but sometimes I need a queue.  Please just make queues more configurable and let users choose for themselves. 

Here's my queue wishlist:

1. Make the batch size for lambda execution configurable (currently defaults to 10)
2. Make long polling the default (I can't think of any downside to this)
3. Make delivery delay configurable
4. Automatically set visibility timeout to 6 times the handler timeout per AWS recommendations and/or make configurable
5.  Allow me to set fifo=false at the @aws level
